backend/lcfs/web/api/allocation_agreement/services.py

```diff
@@ -427,3 +464,49 @@ async def get_compliance_report_by_id(self, compliance_report_id: int):
             )
 
         return compliance_report
+
+    async def calculate_ci_of_fuel(
```

Comment:
question: Should this also be used in update_allocation_agreement where we set the ci_of_fuel when saving the rows? I feel like they should use the same logic.